### PR TITLE
Add memory per job as Torque option

### LINKED ISSUE #5563: Allow setting memory requirements for job for Torque queue
**Is your feature request related to a problem? Please describe.**

The `qsub` command for the Torque compute cluster allows specifying the memory requirements for a job.  Currently, the Torque driver is unable to specify this. This is needed as compute nodes typically have a high cpu-to-mem ratio. An example is a compute node with 120 cores and 450Gb memory. Unless other information is given, the Torque cluster will happily give this compute nodes 120 ERT jobs, leaving only 3GB-ish memory available pr. job. This will fail hard.

**Describe the solution you'd like**
A Torque option in the ert config file

**Describe alternatives you've considered**
The current workaround is to specify that your job requires X number of CPU-cores, implicitly allocating enough memory.

**Additional context**

### PR DESCRIPTION
**Pull request by berland**
**Issue**
Resolves #5563 

**Approach**
* Copy from other PRs that have added options to the Torque driver.
* Utilize new config testing setup using pytest hypothesis to test the added configuration.

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [x] Updated documentation
- [ ] Ensured new behaviour is tested

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

**Comment by berland** [comment]
Try to validate the option value in `QueueConfig`.

**Comment by berland** [comment]
> Try to validate the option value in `QueueConfig`.

Done.

File: docs/reference/configuration/keywords.rst [hunk 1]
```diff
@@ -1748,6 +1748,25 @@ bjobs.
                 QUEUE_OPTION TORQUE NUM_NODES 3
                 QUEUE_OPTION TORQUE NUM_CPUS_PER_NODE 8
 
+.. _torque_memory_per_job:
+.. topic:: MEMORY_PER_JOB
+
+
+        You may indicate to the compute cluster how much memory your job will need to
```
Comment (xjules) [inline docs/reference/configuration/keywords.rst:1755]: `You can specify the amount of memory you will need for running the job on the compute cluster.`  ?

File: src/clib/lib/job_queue/torque_driver.cpp [hunk 2]
```diff
@@ -75,6 +77,7 @@ void *torque_driver_alloc() {
     torque_driver_set_option(torque_driver, TORQUE_QDEL_CMD,
                              TORQUE_DEFAULT_QDEL_CMD);
     torque_driver_set_option(torque_driver, TORQUE_NUM_CPUS_PER_NODE, "1");
+    torque_driver_set_option(torque_driver, TORQUE_MEMORY_PER_JOB, "1g");
```
Comment (xjules) [inline src/clib/lib/job_queue/torque_driver.cpp:80]: Does 1g means default is 1GB?

File: docs/reference/configuration/keywords.rst (hunk 1 above)
Comment (berland) [reply docs/reference/configuration/keywords.rst:1755]: Yes, fixed.

File: src/clib/lib/job_queue/torque_driver.cpp (hunk 2 above)
Comment (berland) [reply src/clib/lib/job_queue/torque_driver.cpp:80]: hmm, yes, but it is probably a bug, it should be "1gb"

File: src/clib/lib/job_queue/torque_driver.cpp [hunk 3]
```diff
@@ -289,6 +303,22 @@ torque_job_type *torque_job_alloc() {
...
+                                  int num_cpus_per_node,
+                                  std::string memory_per_job) {
+    std::string resource_string = "nodes=" + std::to_string(num_nodes);
+
+    if (!cluster_label.empty())
+        resource_string.append(":" + cluster_label);
+
+    resource_string.append(":ppn=" + std::to_string(num_cpus_per_node));
+
+    if (!memory_per_job.empty())
```
Comment (xjules) [inline src/clib/lib/job_queue/torque_driver.cpp:316]: Right, I thought that this line: `torque_driver_set_option(torque_driver, TORQUE_MEMORY_PER_JOB, "1g");`
makes the default to 1GB , so I guess this is not the case.

File: src/clib/lib/job_queue/torque_driver.cpp (hunk 2 above)
Comment (berland) [reply src/clib/lib/job_queue/torque_driver.cpp:80]: It is debated though if we should not provide any default for this.

File: tests/unit_tests/config/config_dict_generator.py [hunk 4]
```diff
@@ -72,6 +72,14 @@ def report_steps(draw):
...
 positives = st.integers(min_value=1, max_value=10000)
 queue_systems = st.sampled_from(["LSF", "LOCAL", "TORQUE", "SLURM"])
+memory_unit = st.sampled_from(["gb", "mb"])
+
+
+@st.composite
+def memory_with_unit(draw):
+    memory_value = draw(positives)
+    unit = draw(memory_unit)
+    return f"{memory_value} {unit}"
```
Comment (berland) [inline tests/unit_tests/config/config_dict_generator.py:82]: remove space!

**Review by xjules — APPROVED** [review]
:rocket:

**Comment by berland** [comment]
Blocked by #5913

**Pull request merged**